inc/search.class.php

```diff
@@ -5388,6 +5388,24 @@ static function addMetaLeftJoin($from_type, $to_type, array &$already_link_table
          return $JOIN;
       }
 
+      if ($from_referencetype === 'Computer' && $to_type === 'OperatingSystem') {
```

Comment:
Why only computer? Better to rely on presence in `$CFG_GLPI['operatingsystem_types']` I gues..

Reply:
Good point, I didn't commit my final version.
Fixed